app/src/main/java/com/duycuong/weather/data/source/local/sqlite/LocationLocalDataSource.java

```diff
@@ -0,0 +1,107 @@
+package com.duycuong.weather.data.source.local.sqlite;
+
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteOpenHelper;
+
+import com.duycuong.weather.data.model.Locationn;
+import com.duycuong.weather.data.source.LocationDataSource;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by DuyCương on 11/02/2018.
+ */
+
+public class LocationLocalDataSource extends SQLiteOpenHelper implements LocationDataSource {
+    public static int VERSION = 1;

```

Comment:
ERROR: [MALICIOUS_CODE][MS_SHOULD_BE_FINAL] com.duycuong.weather.data.source.local.sqlite.LocationLocalDataSource.VERSION isn't final but should be
http://findbugs.sourceforge.net/bugDescriptions.html#MS_SHOULD_BE_FINAL